src/metabase/api/card.clj

```diff
@@ -704,6 +707,113 @@
                  :js-int-to-string?      false}))
 
 
+;;; ------------------------------------------------ Parameters -------------------------------------------------
+(def ^:const result-limit
+  "How many results to return when chain filtering."
+  1000)
+
+(defn- field-clause->field-id
+  "Find the field id in a field clause if it exists.
+
+    (field-clause->field-id [:field 3 nil])
+    ;; -> 3"
+  [field-clause]
+  (mbql.u/match-one field-clause [:field (id :guard integer?) _] id))
+
+(defn- template-tag-target->field-id
+  "Given a template tag target, find the field id that it's connected to.
+  Note that a target has a field id connect to it iff it's a field filter."
+  [card {:keys [id]}]
+  (if-let [template-tags (get-in card [:dataset_query :native :template-tags])]
+    (first (for [[_ template-tag] template-tags
+                  :when (and (= (:id template-tag) id)
+                             (= (:type template-tag) :dimension))
+                  :let  [field-id (field-clause->field-id (:dimension template-tag))]
+                  :when field-id]
+             field-id))
+    (throw (ex-info (tru "Card with ID {0} does not have template tags." (:id card))
+                    {:card-id         (:id card)
+                     :template-tag-id id}))))
+
+(defn- target->field-id
+  "Find the field id that the target is connected to.
+  Target could be a `dimension`, in which target have the shape
+
+    [:dimension [:field 3 nil]]
+
+  Or it could be a `template-tag`, then it should have the shape
+
+    [:template-tag {:id \"6006ad7d-036e-83ec-4d6f-30f82b98ac21\"}]"
+  [card [target-type target-args]]
+  (case target-type
+    :dimension    (field-clause->field-id target-args)
+    :template-tag (template-tag-target->field-id card target-args)
+    nil))
+
+(defn- param-key->field-ids
+ "Get Field ID(s) associated with a parameter in a Card.
+
+    (param-key->field-ids (Card 62) \"ee876336\")
+    ;; -> #{276}"
+  [{:keys [parameter_mappings] :as card} param-key]
+  (into #{} (for [{:keys [target parameter_id]} parameter_mappings
+                  :when (= parameter_id param-key)
+                  :let [field-id (target->field-id card target)]
+                  :when field-id]
```

Comment:
nit: alignment
```suggestion
  (into #{} (for [{:keys [target parameter_id]} parameter_mappings
                   :when (= parameter_id param-key)
                   :let [field-id (target->field-id card target)]
                   :when field-id]
```

Reply:
the original was actually correct, then `when`, `let` keys are not part of the 1st map.